File: old/insultbot/insult_api.py
# insult_api.py
"""Wrapper for https://insult.mattbas.org/api/"""
import requests as req

FILE_FORMAT = "txt"
LINK = f"https://insult.mattbas.org/api/insult.{FILE_FORMAT}"


def get_insult(who="%7Buser%7D", plural=False, template: str = None):
    """
    Requests an insult from the API

    :param who: Who to insult
    :param plural: Whether or not to insult multiple people
    :param template: Template to follow (recommend leaving this as None)
    :return: Formatted insult
    """
    params = {"who": who}
    if template is not None:
        params["template"] = template
    if plural:
        params["plural"] = "on"
    print("Requesting insult from API...")
    response = req.get(LINK, params=params)
    print(f"Received '{response}'")
    return response.content.decode("utf-8").replace("%7Buser%7D", "{user}")  # html handles brackets differently

File: src/insultbot/insultApi.ts
import axios from "axios";

/**
 * Gets an insult from the API found at https://insult.mattbas.org
 *
 * @param who Who to insult
 * @param plural Whether to insult multiple people or not
 * @returns Formatted insult
 */
export default async function getInsult(
  who = "You",
  plural = true
): Promise<string> {
  console.log("Requesting insult from API...");
  const response = await axios.get(
    "https://insult.mattbas.org/api/insult.txt",
    { params: plural ? { who, plural } : { who } }
  );
  console.log(`Received '${response.status} ${response.statusText}'`);
  return response.data;
}
